src/transformers/tokenization_utils.py

```diff
@@ -346,13 +346,18 @@ class PreTrainedTokenizer(PreTrainedTokenizerBase):
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
-        # Added tokens - We store this for both slow and fast tokenizers
-        # until the serialization of Fast tokenizers is updated
-        self.added_tokens_encoder: Dict[str, int] = {}
-        self.added_tokens_decoder: Dict[int, str] = {}
-        self.unique_no_split_tokens: List[str] = []
-        self.tokens_trie = Trie()
+        if self.unk_token is not None and self.unk_token not in self._added_tokens_decoder:
+            if self.convert_tokens_to_ids(self.unk_token) is None:
+                raise ValueError(
+                    "An `unk_token` was set but it is neither part of the `additional_special_tokens` nor is it recognized by the tokenizer."
+                    " Make sure the token exists, whithout it, adding tokens to the model is not possible. Thus the initialization fails."
+                )
+
+        # 2. Add the additional special tokens amd the special tokens if they are not already thre.
+        self._add_tokens(self.all_special_tokens_extended, special_tokens=True)
 
+        # 3. Make sure the Trie has everything in it
+        self._create_trie()
         self._decode_use_source_tokenizer = False
```

Comment:
The logic here is that you cannot initialize a `PreTrainedTokenizerBase` class if you don't have a vocabulary already set. At the class level you define the vocab, the way to convert a token to ids etc.
Once this is done, then only can you initialize the special tokens, because you can know whether or not you have to add them to vocab. 

We make sure that all special tokens are added to the trie when initialising.